libs/base/Data/Vect.idr

```diff
@@ -224,7 +223,7 @@ fromList' {len} ys (x::xs) =
 ||| fromList [1,2,3,4]
 ||| ```
 public export
-fromList : (l : List elem) -> Vect (length l) elem
+fromList : (1 l : List elem) -> Vect (length l) elem
```

Comment:
Why is this function so complicated?

```idris
fromList [] = []
fromList (x :: xs) = x :: fromList xs
```

ought to work! Consequently, I'm not sure what the use case for `fromList'` is.

Reply:
It makes it tail recursive, which I would guess is more important in a strict language like Idris so that you won't bash your stack trying to convert an especially long list to a `Vect`.  Not certain tho – perhaps @edwinb can comment here.

Reply:
We've had this conversation about tail-recursive `public export` functions a few times recently ([#132](https://github.com/idris-lang/Idris2/issues/132#issuecomment-633285153), [#269](https://github.com/idris-lang/Idris2/pull/269#issuecomment-643168588)), so I'll just link what I'd repeat here. :)

There are issues with what I suggest, though, as @gallais pointed out, so take it just as an illustration why making everything tail recursive is not automatically a good thing.